internal/runtime/client/client.go

```diff
@@ -549,3 +565,21 @@ func defaultDiscoveryResponse(discovery *runtimehooksv1.DiscoveryResponse) *runt
 	}
 	return discovery
 }
+
+// matchNamespace returns true if the passed namespace matches the selector. It returns an error if the namespace does
+// not exist in the API server.
+func (c *client) matchNamespace(ctx context.Context, selector labels.Selector, namespace string) (bool, error) {
+	ns := &metav1.PartialObjectMetadata{}
+	ns.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "",
+		Version: "v1",
+		Kind:    "Namespace",
+	})
+	if err := c.kubeClient.Get(ctx, ctrlclient.ObjectKey{Name: namespace}, ns); err != nil {
+		return false, err
+	}
+	if !selector.Matches(labels.Set(ns.GetLabels())) {
+		return false, nil
+	}
+	return true, nil
```

Comment:
```suggestion
	return selector.Matches(labels.Set(ns.GetLabels())), nil
```
nit, I think this is a bit simpler